dsn/src/builder.rs

```diff
@@ -244,48 +227,14 @@ impl Dsn {
         let farmer_readers_and_pieces = Arc::new(parking_lot::Mutex::new(None));
         let protocol_version = hex::encode(client.info().genesis_hash);
         let farmer_piece_cache = Arc::new(parking_lot::RwLock::new(None));
-        let maybe_farmer_local_record_provider =
-            MaybeLocalRecordProvider::new(farmer_piece_cache.clone());
+        let local_records_provider = MaybeLocalRecordProvider::new(farmer_piece_cache.clone());
```

Comment:
Is this wrapper still needed? Cache can be used directly as a provider now.

Reply:
This wrapper is required since the farmer piece cache is behind `Arc<RwLock<Option`.

Reply:
Why does it need to be behind `Arc<RwLock<Option<T>>>` though?

Reply:
This DSN instance is shared between node and farmer. And it is created when node has started. So, generally objects owned by farmer are initialized as `None` behind `RwLock` when the instance has been created. 

Once farmer has started, it replaces that `None` with actual piece cache.